src/main/scala/scorex/api/http/AddressApiRoute.scala

```diff
@@ -213,8 +238,17 @@ case class AddressApiRoute(settings: RestAPISettings, wallet: Wallet, state: Sta
       acc.address,
       confirmations,
       state.accountPortfolio(acc).balance
-    )))
-      .getOrElse(InvalidAddress)
+    ))).getOrElse(InvalidAddress)
+  }
+
+  private def balancesDetailsJson(account: Account): BalanceDetails = {
+    val portfolio = state.accountPortfolio(account)
+    BalanceDetails(
+      account.address,
+      portfolio.balance,
+      consensusModule.generatingBalance(account, state.height)(transactionModule),
+      portfolio.balance - portfolio.leaseInfo.leaseOut,
+      state.effectiveBalance(account))
```

Comment:
jsut FYI, `portfolio already contains all the info, it's plain `lazy val effectiveBalance`